### PR TITLE
chore(atomic): update i18next (#2808

### PR DESCRIPTION
**Pull request by btaillon**
https://coveord.atlassian.net/browse/KIT-2334

I spent a while trying to add typing for localization following the new docs (https://www.i18next.com/overview/typescript). Unfortunately, I kept getting `TS2589: Type instantiation is excessively deep and possibly infinite.` when `i18n.t` was called with a `ns` in its options parameter. After a while, I decided it wasn't worth the effort and removed typing by declaring `resources` as a string record.
* If you wish to see what this looks like or reproduce the issue, I created a branch here: https://github.com/coveo/ui-kit/pull/2809

To do before merging:
- [x] Verify that exported type definitions aren't significantly changed.
- [x] Verify that plurals aren't broken.

### CODE REVIEW

File: packages/atomic/src/components/common/interface/i18n.ts [hunk 1]
```diff
@@ -61,5 +61,12 @@ export function initi18n(atomicInterface: BaseAtomicInterface<AnyEngineType>) {
     interpolation: {
       escape: (str) => DOMPurify.sanitize(str),
     },
+    compatibilityJSON: 'v3',
```
Comment (olamothe) [inline packages/atomic/src/components/common/interface/i18n.ts:64]: If I understand correctly from i18n doc, this is what makes plural support the same as the version currently used on master, is that correct ?

File: packages/atomic/src/locales.json [hunk 2]
```diff
@@ -1212,6 +1212,14 @@
     "en": "Apply custom start and end dates for the {{label}} facet",
     "fr": "Appliquer les dates de début et de fin manuelles pour la facette {{label}}"
   },
+  "past-minute": {
```
Comment (olamothe) [inline packages/atomic/src/locales.json:1215]: Why is this needed now ?

File: packages/atomic/src/components/common/interface/i18n.ts [hunk 3]
```diff
@@ -61,5 +61,12 @@ export function initi18n(atomicInterface: BaseAtomicInterface<AnyEngineType>) {
...
   });
 }
+
+declare module 'i18next' {
+  interface CustomTypeOptions {
+    resources: Record<string, Record<string, string>>;
```
Comment (olamothe) [inline packages/atomic/src/components/common/interface/i18n.ts:70]: Where is this type used ?

File: packages/atomic/src/components/common/interface/i18n.ts (hunk 1 above)
Comment (btaillon) [reply packages/atomic/src/components/common/interface/i18n.ts:64]: Yes

File: packages/atomic/src/locales.json (hunk 2 above)
Comment (btaillon) [reply packages/atomic/src/locales.json:1215]: I forgot to mention that.

While testing with typings, I noticed that this was missing. It's not strictly needed for the PR.

File: packages/atomic/src/components/common/interface/i18n.ts (hunk 3 above)
Comment (btaillon) [reply packages/atomic/src/components/common/interface/i18n.ts:70]: It's used by `i18n.t(...)`.

**Review by olamothe — APPROVED** [review]

File: packages/atomic/src/components/common/interface/i18n.ts (hunk 3 above)
Comment (olamothe) [reply packages/atomic/src/components/common/interface/i18n.ts:70]: So, is it really useful then ? Or mostly an artefact of trying to add typing to the available keys ?

Comment (btaillon) [reply packages/atomic/src/components/common/interface/i18n.ts:70]: Yeah, we need it to compile, since i18next now tries to add typings by default.
